### PR TITLE
Add 6pac-slickgrid@2.3.7 w/ git auto-update

### PR DESCRIPTION
**Pull request by sufuf3**
cc #10036, cc @6pac , @mleibman
@kennynaoh please help me review this PR, thank you.

Pull request for issue: #10036 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/6pac/SlickGrid
 * Official website (optional, not the repository): http://wiki.github.com/6pac/SlickGrid
 * NPM package url (optional): https://www.npmjs.com/package/slickgrid
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 61
   - Count of stars: 382
   - Count of forks: 2011
 * NPM download stats (optional):
   - Downloads in the last day: 98
   - Downloads in the last week: 404
   - Downloads in the last month: 1579

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by 6pac** [comment]
Before we proceed, are you aware that the npm versions I have been provided are now part of the 6pac repo (in the ```npm``` folder) and are registered under the 'slickgrid' tag ?

**Comment by sufuf3** [comment]
@6pac sorry about I add wrong information at the first comment.
I modified it.
And I add the right npm version at this pr. (from https://www.npmjs.com/package/slickgrid)
Thank you.

**Comment by 6pac** [comment]
you are right that the npm folder versions are slightly older. i'm in the process of working out builds for nuget and bower as well. once that's in black i've created a script to update all the denormalised files. just got to start remembering to use it every time i push. that looks like it's leading to a push script too ...

**Comment by sufuf3** [comment]
@maruilian11 thank you for your review and suggestion.
I modified it, and please help me review this PR again, thank you.

File: ajax/libs/slickgrid/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "license": "MIT",
+  "homepage": "https://github.com/mleibman/SlickGrid/wiki",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/6pac/SlickGrid.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "lib/*",
```
Comment (maruilian11) [inline ajax/libs/slickgrid/package.json]: We only add libs are not on cdnjs and it doesn't has git repo or npm package if it is needed for this lib.

File: ajax/libs/slickgrid/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          "slick.core.js"
```
Comment (maruilian11) [inline ajax/libs/slickgrid/package.json]: Do we only need this file? How about other `slick*`?

File: ajax/libs/slickgrid/package.json (hunk 1 above)
Comment (sufuf3) [inline ajax/libs/slickgrid/package.json]: In lib, the CDNJS didn't hold `jquery.simulate.js`, `jquery.event.drag-2.2.js`,` jquery.event.drag-2.3.0.js`,` jquery.event.drop-2.2.js`, `jquery.event.drop-2.3.0.js`

And I saw some people use lib in their code form this lib. eg. https://github.com/person2865/d3js-practise/blob/76051d072edfe9af0bd6de75eab52f20967e41f7/index.html

so, I think it need this lib.

PS1. For` jquery-simulate` maybe user can use from https://github.com/jquery/jquery-simulate
PS2. For `jquery.event.drag` `jquery.event.drop `maybe user can use from http://threedubmedia.com/ , but in [README ](https://github.com/6pac/SlickGrid/blob/master/README.md) it said this lib updated jquery.event.drag and updated jquery.event.drop to be compatible with jQuery 3

Comment (sufuf3) [inline ajax/libs/slickgrid/package.json]: @maruilian11  In lib, the CDNJS didn't hold `jquery.simulate.js`, `jquery.event.drag-2.2.js`,`jquery.event.drag-2.3.0.js`,`jquery.event.drop-2.2.js`, `jquery.event.drop-2.3.0.js`
should I include them?

Comment (maruilian11) [inline ajax/libs/slickgrid/package.json]: if cdnjs doesn't hold those libs, we should host them, too. If they has git repo or npm package, send another PR to add them, and mention this issue in the PR. If they don't adding them with this lib is fine.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by sufuf3** [comment]
@cdnjs/intern3 please help me review this PR, thanks.
Note:
The files under `lib` I just add `lib/jquery.event.*`, and send another PR add `jquery-simulate`, other files can get from CDNJS, so I not include them.
Why I don't send another PR to add `"jquery.event.*"`? Because I can't find the library, and I think these files are edited by the author, so include in this library.

**Comment by sufuf3** [comment]
@extend1994 
Here are the examples --> https://github.com/6pac/SlickGrid/tree/master/examples and https://github.com/6pac/SlickGrid/wiki/Examples
* `"slick*"`
As you can see, many `html` files use `slick.*` and dependent on `lib`.
* `"lib/jquery.event.*"`
We only add libs are not on cdnjs and it doesn't have git repo or npm package if it is needed for this lib.
CDNJS include most of the files under `lib`, but `lib/jquery.event*` and `jquery-simulate`.
* `"images/*"`
Some of the `*.js` and `*.css` files use the files under `images`, so I add them.

File: ajax/libs/slickgrid/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "license": "MIT",
+  "homepage": "https://github.com/mleibman/SlickGrid/wiki",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/6pac/SlickGrid.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "lib/jquery.event.*",
```
Comment (hare1039) [inline ajax/libs/slickgrid/package.json]: I think the library `jquery.event.drag` is this one `https://github.com/devongovett/jquery.event.drag`
or this one
`https://github.com/threedubmedia/jquery.threedubmedia`
So I think we don't need to include `lib/jquery.event.*`

Comment (sufuf3) [inline ajax/libs/slickgrid/package.json]: @hare1039 thank you! I find the `jquery.event.*` is from https://github.com/threedubmedia/jquery.threedubmedia . Ref: https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js#L4
https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js#L4

I'll add this library later.
And delete this line.

**Comment by sufuf3** [comment]
@hare1039 I deleted this line and remove the folder which is `lib`.
Please help me review this PR, thanks.

**Comment by sufuf3** [comment]
@hare1039 Oh, no.
I think we should still host the `lib/jquery.event.*` because the last updated of the files in origin library is 2012-05-21.
But the last updated of the `jquery.event.*` files in 6pac/SlickGrid is 2016-08-16. 

ref: https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js#L8
https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js#L8

Because I readded the line and readded the `lib` folder, so please help me review this PR again, thanks.

File: ajax/libs/slickgrid/package.json (hunk 3 above)
Comment (sufuf3) [inline ajax/libs/slickgrid/package.json]: go to https://github.com/cdnjs/cdnjs/pull/10444#issuecomment-298173745

**Comment by 6pac** [comment]
Yep, ```jquery.event.drag-2.3.0.js``` and ```drop``` were updates for jQuery 2.x, 3.x

They are not available anywhere else (these were our own updates, not release by ThreeDub)

**Comment by sufuf3** [comment]
@6pac Thank you very much.

**Comment by sufuf3** [comment]
@hare1039 @clairetsai818 please help me review this PR, thank you!

**Comment by sufuf3** [comment]
ping @hare1039 @clairetsai818

**Comment by sufuf3** [comment]
@maruilian11 Please help me review this PR, Thank you.

File: ajax/libs/slickgrid/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          "images/*",
```
Comment (pvnr0082t) [inline ajax/libs/slickgrid/package.json]: I think it will be better to use `"images/*.+(gif|png)"`

**Comment by sufuf3** [comment]
@pvnr0082t I modified it.
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@sufuf3 Don't we add `jquery.event.drag.js` and `jquery.event.drop.js` as another lib?

**Comment by sufuf3** [comment]
@maruilian11 although the `jquery.event.*` is from https://github.com/threedubmedia/jquery.threedubmedia 
 [1][2], as the author said https://github.com/cdnjs/cdnjs/pull/10444#issuecomment-298275674 it modified by this library, and it not as same with origin `jquery.event.*`[3][4].

Ref: 
[1] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js#L4
[2] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js#L4
[3] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js#L8
[4] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js#L8

**Comment by maruilian11** [comment]
Thanks for explaining. Don't we need `lib/firebugx.js` and `plugins/*`?

**Comment by sufuf3** [comment]
@maruilian11  you're right, I think I should add them.
 I added `plugins/*` and `lib/firebugx.js` , minified these files, and modified the `package.json` add lines as followings:
```js
         "lib/firebugx*",
         "plugins/slick*.+(js|css|map)"
```

Please help me review this PR, Thank you.

File: ajax/libs/slickgrid/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "target": "git://github.com/6pac/SlickGrid.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "lib/jquery.event.*",
+          "lib/firebugx*",
+          "images/*.+(gif|png)",
+          "slick*",
+          "plugins/slick*.+(js|css|map)"
```
Comment (maruilian11) [inline ajax/libs/slickgrid/package.json]: Let's also use `plugins/slick*` here, thanks.

Comment (sufuf3) [inline ajax/libs/slickgrid/package.json]: May I ask what situation should use `*` or `+(js|css|map)`?
Why the files under `images` should use `"images/*.+(gif|png)"`, and why the files under `plugins` should use `"plugins/slick*"`?

Comment (PeterDaveHello) [inline ajax/libs/slickgrid/package.json]: It's both fine.

**Comment by maruilian11** [comment]
@pvnr0082t May I ask why you think using `images/*.+(gif|png)` is better than `image/*`? Thanks.

**Comment by PeterDaveHello** [comment]
@maruilian11 @pvnr0082t looks both fine to me unless there is something we need to exclude here.

**Comment by maruilian11** [comment]
@PeterDaveHello I think `images/*` can avoid we losing some images which are not `gif` or `png`. We may need to send another PR to fix it if it happen, and also it may take times to do that. Since we usually host most type of img, I think change back to use `image/*` will feel more comfortable. Thanks.

**Comment by sufuf3** [comment]
So, if there is something we need to exclude here we need to use `+(...|...)`, otherwise we use `*` is fine, isn't it?

**Comment by PeterDaveHello** [comment]
@sufuf3 something like that, it really depends.

**Comment by sufuf3** [comment]
I can modify it, to let it be better, should I modify as followings?
`images/*.+(gif|png)` -> `images/*`
`plugins/slick*.+(js|css|map)` -> `plugins/slick*`

**Comment by PeterDaveHello** [comment]
@maruilian11 @pvnr0082t you guys have suggestion for @sufuf3 ?

**Comment by sufuf3** [comment]
I think I'll modify as the followings 
```
"lib/jquery.event.*",
"lib/firebugx*",
```
to 
```
"lib/+(jquery.event.*|firebugx*)",
```

But I have no idea about the followings:
`images/*.+(gif|png)` -> `images/*`
`plugins/slick*.+(js|css|map)` -> `plugins/slick*`
If I get what should I do, I'll fix it.

**Comment by pvnr0082t** [comment]
It's okay.

**Comment by sufuf3** [comment]
@pvnr0082t should I modify from `images/*.+(gif|png)` to `images/*` and from `plugins/slick*.+(js|css|map)` to `plugins/slick*`?

**Comment by PeterDaveHello** [comment]
@sufuf3 you should label the issues not PRs.

**Comment by maruilian11** [comment]
```
images/*.+(gif|png) -> images/*
plugins/slick*.+(js|css|map) -> plugins/slick*
```

I think it will be great if modify them, thanks!

**Comment by sufuf3** [comment]
@maruilian11 @pvnr0082t I modified the `package.json`.
Please help me review this PR again, Thank you.

**Comment by sufuf3** [comment]
@maruilian11 since the `mleibman/SlickGrid` said `This repo hasn't been updated in a while. https://github.com/6pac/SlickGrid/wiki seems to be the most active fork at the moment.`
ref: https://github.com/mleibman/SlickGrid#welcome-to-slickgrid

I think we can host it.
If it's necessary I can add `mleibman/SlickGrid` also.

**Comment by maruilian11** [comment]
@PeterDaveHello Do we add both repos in this case?

**Comment by PeterDaveHello** [comment]
Depends on their popularity, simply.

**Comment by maruilian11** [comment]
@sufuf3 Let's add both of them, thanks!

**Comment by sufuf3** [comment]
@maruilian11 OK I'll send another PR to add `mleibman/SlickGrid`.
In this PR, is it good?

**Comment by maruilian11** [comment]
@sufuf3 Because this is the forked one, don't you think we need to modify the package name?

**Comment by sufuf3** [comment]
@maruilian11 Yes, you're right.
May I rename it to `6pac-slickgrid`?

**Comment by sufuf3** [comment]
@maruilian11 I renamed the package name.
Please help me review this PR, Thank you.

File: ajax/libs/6pac-slickgrid/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  "keywords": [
+    "slickgrid",
+    "grid"
+  ],
+  "author": {
+    "name": "Michael Leibman",
+    "email": "michael.leibman@gmail.com"
+  },
+  "license": "MIT",
+  "homepage": "https://github.com/mleibman/SlickGrid/wiki",
```
Comment (maruilian11) [inline ajax/libs/6pac-slickgrid/package.json]: Doesn't it need to be changed?

**Comment by sufuf3** [comment]
@maruilian11 I modified the `homepage` please help me review this PR, thanks.

File: ajax/libs/6pac-slickgrid/package.json (hunk 6 above)
Comment (PeterDaveHello) [inline ajax/libs/6pac-slickgrid/package.json]: @maruilian11 be more clear next time.

**Comment by PeterDaveHello** [comment]
Can we try to host jquery.evert* individually first?

**Comment by sufuf3** [comment]
I'll add library `jquery.evert*` from https://github.com/threedubmedia/jquery.threedubmedia
But as the author said https://github.com/cdnjs/cdnjs/pull/10444#issuecomment-298275674 I think we still host the `jquery.evert*` in this library.


Note:
As the author said #10444 (comment) these files modified by this library, and it not as same with origin jquery.event.*[1][2] any more.

Ref:
[1] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js#L8
[2] https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js#L8

**Comment by sufuf3** [comment]
@PeterDaveHello 
Could you please give me some tips that what should I do with host jquery.evert* individually?

Should I add another library jquery.evert*(v2.2) from https://github.com/threedubmedia/jquery.threedubmedia
and create another two PR (jquery.evert.drop@v2.3.0 and jquery.evert.drag@v2.3.0) from 6pac?

Then there are three libraries as the followings:
1. [jquery.threedubmedia](https://github.com/threedubmedia/jquery.threedubmedia/releases)  (there is one git tag and the latest realease on 21 May 2012)
2. [6pac-jquery.evert.drop](https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drop-2.3.0.js) (manually add v2.3.0)
2. [6pac-jquery.evert.drag](https://github.com/6pac/SlickGrid/blob/master/lib/jquery.event.drag-2.3.0.js) (manually add v2.3.0)

Thank you.

**Comment by PeterDaveHello** [comment]
If jquery.evert.drop v2.3.0 is a forked/modified version only exist on 6pac I won't host it individually, hosting jquery.threedubmedia is good.

**Comment by sufuf3** [comment]
@cdnjs/intern3 
Please help me review this PR, Thank you.

I send a PR to add  jquery.evert*(v2.2) #11422 
And modified from `"lib/+(jquery.event.*|firebugx*)"` to `"lib/+(jquery.event.*-!(2.2)|firebugx*)"`
And update version from v2.3.6 to v2.3.7

**Comment by sufuf3** [comment]
@extend1994 @dakshshah96 @hare1039 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@extend1994 @dakshshah96 @hare1039
Please help me review this PR, Thank you.

**Comment by uira** [comment]
@sufuf3 @seenzah consegue corrigir isso?

**Comment by Gianfranco97** [comment]
@uira Of course, let me change things so I can go directly there

**Comment by maruilian11** [comment]
@sufuf3 Could you please tell me why we don't fetch `jquery.event.drag-2.2.js`? And I think it's better to use `gitignore` to ignore them.

**Comment by sufuf3** [comment]
@maruilian11 
The `jquery.event.drag-2.2.js` and `jquery.event.drop-2.2.js`, CDNJS will host. PR: #11422
as this comment said: https://github.com/cdnjs/cdnjs/pull/10444#issuecomment-305602046

**Comment by maruilian11** [comment]
@sufuf3 Thank you
```
I'll add library jquery.evert* from https://github.com/threedubmedia/jquery.threedubmedia
But as the author said #10444 (comment) I think we still host the jquery.evert* in this library.
```
--->
```
I'll add library jquery.event* from https://github.com/threedubmedia/jquery.threedubmedia
But as the author said #10444 (comment) I think we still host the jquery.event*2.3.0* in this library.
```
will be more clear

Let's use `.gitignore` to avoid fetching `jquery.event.*-2.2.js`?

**Comment by sufuf3** [comment]
@maruilian11 I'll update the `.gitignore`.
May I ask that should I revise `"lib/+(jquery.event.*-!(2.2)|firebugx*)"` to 
origin  `"lib/+(jquery.event.*|firebugx*)"` ?

**Comment by maruilian11** [comment]
@sufuf3 Yes, that's why we update the `.gitignore`.

**Comment by sufuf3** [comment]
@maruilian11 I update the `.gitingnore` and review the filemap from `"lib/+(jquery.event.*-!(2.2)|firebugx*)"` to `"lib/+(jquery.event.*|firebugx*)"`
please help me review the PR, thnak you.

**Comment by sufuf3** [comment]
@extend1994 @pvnr0082t 
Please help me review this PR, Thank you.

**Comment by extend1994** [comment]
LGTM
@PeterDaveHello please help review this pull request, thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello 
Please help me review this PR, Thank you.
